apps/web/app/my-reviews/page.tsx

```diff
@@ -0,0 +1,141 @@
+"use client";
+
+import { MyReviewCard } from "@components/ui/cards/my-review-card";
+import SessionModal from "@components/ui/session-modal";
+import { Accordion, Badge, Center, Grid, Loader, Paper, Stack, Text } from "@mantine/core";
+import { useDisclosure } from "@mantine/hooks";
+import { APPROVED, BASE_API_ENDPOINT, ERROR, INFO, PENDING, REJECTED, SUCCESS } from "@utils/constants";
+import fetcher from "@utils/fetcher";
+import { useAuth } from "hooks/use-auth";
+import { useEffect, useState } from "react";
+import type { AccordionItems, MyReviewsData } from "types";
+import type { Session } from "@supabase/supabase-js";
+
+const initialEmptyPosts: MyReviewsData[] = [];
+
+export default function MyReviews(): JSX.Element {
+	const [sessionData, setSessionData] = useState<Session | null>();
+	const [isLoading, setIsLoading] = useState<boolean>(true);
+	const [myReviewsData, setMyReviewsData] = useState<MyReviewsData[] | null>([]);
+
+	const { getSession } = useAuth();
+	const [opened, { open: openSessionModal, close: closeSessionModal }] = useDisclosure(false);
+
+	useEffect(() => {
+		if (!sessionData) {
+			setIsLoading(true);
+			getSession()
+				.then((session) => {
+					setSessionData(session);
+					if (!session) openSessionModal();
+				})
+				.catch(console.error)
+				.finally(() => {
+					setIsLoading(false);
+				});
+		}
+	}, [getSession, openSessionModal, sessionData]);
+
+	useEffect(() => {
+		if (!sessionData) {
+			return;
+		}
+		setIsLoading(true);
+
+		const fetchMyReviews = async (): Promise<void> => {
+			try {
+				const results = await fetcher<{ data: MyReviewsData[] }>(
+					`${BASE_API_ENDPOINT}/myreviews`,
+					sessionData.access_token || ""
+				);
+
+				setMyReviewsData(results.data?.data || []);
+			} catch (err) {
+				console.error(err);
+				// Handle error here
+			} finally {
+				setIsLoading(false);
+			}
+		};
+
+		fetchMyReviews().catch(console.error);
+	}, [sessionData]);
+
+	const approvedPosts: MyReviewsData[] = myReviewsData
+		? myReviewsData.filter((item) => item.status === APPROVED)
+		: initialEmptyPosts;
+	const pendingPosts: MyReviewsData[] = myReviewsData
+		? myReviewsData.filter((item) => item.status === PENDING)
+		: initialEmptyPosts;
+	const rejectedPosts: MyReviewsData[] = myReviewsData
+		? myReviewsData.filter((item) => item.status === REJECTED)
+		: initialEmptyPosts;
+
+	const filteredCourseCards: AccordionItems[] = [
+		{
+			value: APPROVED,
+			posts: approvedPosts.length !== 0 ? approvedPosts : [],
```

Comment:
nit picking: Are these ternaries still needed? I thought the case of empty arrays would already be handled by: 
```ts
const initialEmptyPosts: MyReviewsData[] = [];

const approvedPosts: MyReviewsData[] = myReviewsData
 	? myReviewsData.filter((item) => item.status === APPROVED)
 	: initialEmptyPosts;
const pendingPosts: MyReviewsData[] = myReviewsData
 	? myReviewsData.filter((item) => item.status === PENDING)
 	: initialEmptyPosts;
 const rejectedPosts: MyReviewsData[] = myReviewsData
 	? myReviewsData.filter((item) => item.status === REJECTED)
 	: initialEmptyPosts;
```

Reply:
Exactly. It's extra code. I'll fix it